### PR TITLE
OCPBUGS-37486: Set right endpointSlice port

### PR DESCRIPTION
**Pull request by enxebre**
**What this PR does / why we need it**:
This https://github.com/openshift/hypershift/pull/4228 introduced an inconsistency between the endpoint and endpointSlice ports. Resulting in old HCs having hcp.Spec.Networking.APIServer.Port=443 failing to talk to the CP via kubernetes.svc.
This PR fixes that.

**Which issue(s) this PR fixes** *(optional, use `fixes #<issue_number>(, fixes #<issue_number>, ...)` format, where issue_number might be a GitHub issue, or a Jira story*:
Fixes #

**Checklist**
- [ ] Subject and description added to both, commit and PR.
- [ ] Relevant issues have been referenced.
- [ ] This change includes docs. 
- [ ] This change includes unit tests.

### CODE REVIEW

**Comment by enxebre** [comment]
/hold
to add unit test

**Comment by enxebre** [comment]
cc @rtheis

File: control-plane-operator/hostedclusterconfigoperator/controllers/resources/resources_test.go [hunk 1]
```diff
@@ -1022,3 +1024,59 @@ func compareICSAndIDMS(g *WithT, ics []hyperv1.ImageContentSource, idms *configv
...
+						APIServer: &hyperv1.APIServerNetworking{
+							Port: ptr.To(int32(443)),
+						},
+					},
+				},
+			},
+			expectedPort: int32(443),
+		},
+		{
+			name: "When HC has no hcp.spec.networking.apiServer.port set, endpoint and slice should have port 6443",
```
Comment (muraee) [inline control-plane-operator/hostedclusterconfigoperator/controllers/resources/resources_test.go:1049]:
```suggestion
			name: "When HC has no hcp.spec.networking.apiServer.port not set, endpoint and slice should have port 6443",
```

**Comment by enxebre** [comment]
--- FAIL: TestValidateKeyPairItempotency (0.51s)
    --- FAIL: TestValidateKeyPairItempotency/0 (0.32s)
        tls_test.go:168: validation failed when config was unchanged: actual validity 74304.17291666666 differs from expected 74304.17291219337

/test unit

**Review by rtheis — APPROVED** [review]
/lgtm

Sorry about causing this problem.  Thank you for fixing it.

**Pull request merged**